examples/log-mpi.py

```diff
@@ -10,6 +11,24 @@
 from mpi4py import MPI
 
 
+class UserLogQuantity(LogQuantity):
+    """Logging support for arbitrary user quantities."""
+
+    def __init__(self, name, value=None, unit=None,
+          description=None) -> None:
+        LogQuantity.__init__(self, name=name, unit=unit,
+                             description=description)
+        self._quantity_value = value
+
+    def __call__(self) -> float:
+        """Return the actual logged quantity."""
+        return self._quantity_value
```

Comment:
```suggestion
        val = self._quantity_value
        self._quantity_value = None
        return val
```
(to avoid gathering stale values)